src/components/Warehouse/WarehouseCard.js

```diff
@@ -3,14 +3,80 @@ import RightChevron from "../../assets/icons/chevron_right-24px.svg";
 import DeleteIcon from "../../assets/icons/delete_outline-24px.svg";
 import EditIcon from "../../assets/icons/edit-24px.svg";
 import { useNavigate } from "react-router-dom";
+import axios from "axios";
+import DeleteWarehouse from "../Dialog/DeleteWarehouse";
+import { useRef, useState, useEffect } from "react";
 
 const WarehouseCard = ({ item }) => {
   const navigate = useNavigate();
+  const [warehouses, setWarehouses] = useState([]);
+  const id = item.id;
+  const idWarehouseRef = useRef();
+  const [dialog, setDialog] = useState({
+    warehouseName: "",
+    isLoading: false,
+  });
 
   const handleWarehouseClick = (id) => {
     navigate(`/warehouses/${id}/inventories`);
   };
 
+  useEffect(() => {
+    const fetchWarehouse = async () => {
+      try {
+        const resp = await axios.get(
+          `http://localhost:8080/api/warehouses/${id}`
+        );
+        setWarehouses(resp.data);
+      } catch (error) {
+        console.error("Error fetching a warehouse data:", error);
+      }
+    };
+    fetchWarehouse();
+  }, [id]);
+
+  const handleDialog = (warehouseName, isLoading) => {
+    setDialog({
+      warehouseName,
+      isLoading,
+    });
+  };
+
+  const deleteWarehouse = async () => {
+    let response = null;
+    try {
+      response = await axios.delete(
+        `http://localhost:8080/api/warehouses/${id}`
+      );
+    } catch (err) {
+      console.error(err);
+    }
+    if (response.data) {
+      return response.data;
```

Comment:
Since you are deleting I am not expecting any data to be returned. We should just looking for the status code and show some sort of message on successful delete